middleware/userFnLoader.ts

```diff
@@ -1,17 +1,68 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-var-requires */
 
-export default function loadUserFunction(packageName: string): any {
-  if (!packageName) {
-    throw `Could not locate salesforce function module: package name not defined.`;
+import Module =  require('module');
+import path = require('path');
+
+/**
+ * Get the exported function from the user's module
+ * @param mod - The user's module
+ * @return function
+ */
+const getUserFn = function(mod: any): Function {
+  if (mod.__esModule && typeof mod.default === 'function') {
+    return mod.default;
   }
-  let mod;
+  return mod;
+};
+
+/**
+ * Build an enriched function - Wraps the user function to provide enhanced
+ * arguments (initialized InvocationEvent, Context, and Logger). Will not wrap
+ * the function if a modern salesforce-sdk is not installed.
+ *
+ * @param userFn - The function exported by the user.
+ * @param userMod - The user's module that contains the function. Used to lookup
+ *   the salesforce-sdk that may have been installed by the user.
+ * @return enrichedFn - The users function, which may be wrapped.
+ */
+const enrichFn = function(userFn: Function, userModName: string): Function {
+  let sdk, errmsg;
   try {
-    mod = require(packageName);
+    const sdkPath = path.join(userModName, 'node_modules', "@salesforce/salesforce-sdk");
+    sdk = require(sdkPath);
   } catch (err) {
-    throw `Could not locate salesforce function module: ${err}`;
+    console.error(err);
```

Comment:
Remove this

Reply:
The logger we create in the middleware is request specific and this happens prior to any request (it happens in build as well as when the function is first imported by riff). I think I'm tempted to leave this as-is.

Reply:
I tend to agree with leaving it as-is. This message will happen during function startup, so it'll be obvious, and we won't have the requestID at that point anyway.